src/python/dxpy/utils/describe.py

```diff
@@ -566,19 +571,76 @@ def print_app_desc(desc, verbose=False):
+
+    print_field("ID", desc["id"])
+    print_field("Class", desc["class"])
+    if 'billTo' in desc:
+        print_field("Billed to", desc['billTo'][5 if desc['billTo'].startswith('user-') else 0:])
+    print_field("Name", desc["name"])
+    print_field("Version", desc["version"])
+    print_list_field("Aliases", desc["aliases"])
+    print_field("Created by", desc["createdBy"][5 if desc['createdBy'].startswith('user-') else 0:])
+    print_field("Created", render_timestamp(desc['created']))
+    print_field("Last modified", render_timestamp(desc['modified']))
+    # print_json_field('Open source', desc['openSource'])
+    # print_json_field('Deleted', desc['deleted'])
+    if not desc.get('deleted', False):
+        advanced_inputs = []
+        if 'published' not in desc or desc["published"] < 0:
+            print_field("Published", "-")
+        else:
+            print_field("Published", render_timestamp(desc['published']))
+        if "title" in desc and desc['title'] is not None:
+            print_field("Title", desc["title"])
+        if "subtitle" in desc and desc['subtitle'] is not None:
+            print_field("Subtitle", desc["subtitle"])
+        if 'summary' in desc and desc['summary'] is not None:
+            print_field("Summary", desc['summary'])
+        print_list_field("Categories", desc["categories"])
+        if 'details' in desc:
+            print_json_field("Details", desc["details"])
+        print_field("API version", desc["dxapi"])
+
+        # Additionally, print inputs, outputs, stages of the underlying workflow
+        # from the region of the current workspace
+        current_project = dxpy.WORKSPACE_ID
+        if current_project:
+            region = dxpy.api.project_describe(current_project, input_params={"fields": {"region": True}})["region"]
+            if region and region in desc['regionalOptions']:
+                workflow_desc = desc['regionalOptions'][region]['workflowDescribe']
+                print_field("Workflow region", region)
+                if 'id' in workflow_desc:
+                    print_field("Workflow ID", workflow_desc['id'])
+                if workflow_desc.get('inputSpec') is not None and workflow_desc.get('inputs') is None:
+                    print_nofill_field("Input Spec", get_io_spec(workflow_desc['inputSpec'], skip_fields=get_advanced_inputs(workflow_desc, verbose)))
+                if workflow_desc.get('outputSpec') is not None and workflow_desc.get('outputs') is None:
+                    print_nofill_field("Output Spec", get_io_spec(workflow_desc['outputSpec']))
+                if  workflow_desc.get('inputs') is not None:
+                    print_nofill_field("Workflow Inputs", get_io_spec(workflow_desc['inputs']))
+                if  workflow_desc.get('outputs') is not None:
+                    print_nofill_field("Workflow Outputs", get_io_spec(workflow_desc['outputs']))
+                if 'stages' in workflow_desc:
```

Comment:
Minor, may want to add a note somewhere in the describe if there is an 'inputs' and 'outputs' field, smth like the following: 'Note: this workflow has an explicit input and output specification, and as such stage inputs cannot be modified at run-time.'